packages/editor/src/components/page-attributes/test/order.js

```diff
@@ -1,100 +1,85 @@
 /**
  * External dependencies
  */
-import { mount } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 /**
  * Internal dependencies
  */
 import { PageAttributesOrder } from '../order';
 
 describe( 'PageAttributesOrder', () => {
-	it( 'should reject invalid input', () => {
-		const onUpdateOrder = jest.fn();
-		const wrapper = mount(
-			<PageAttributesOrder onUpdateOrder={ onUpdateOrder } />
-		);
-
-		wrapper.find( 'input' ).simulate( 'change', {
-			target: {
-				value: 'bad',
-			},
-		} );
-
-		wrapper.find( 'input' ).simulate( 'change', {
-			target: {
-				value: '----+++',
-			},
-		} );
-
-		wrapper.find( 'input' ).simulate( 'change', {
-			target: {
-				value: '-',
-			},
+	/**
+	 * When starting to type inside the spinbutton, select the current value
+	 * in order to override it with the new value afterwards.
+	 */
+	const typeOptions = {
+		initialSelectionStart: 0,
+		initialSelectionEnd: 1,
+	};
+
+	it( 'should reject invalid input', async () => {
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
 		} );
 
-		wrapper.find( 'input' ).simulate( 'change', {
-			target: {
-				value: '+',
-			},
-		} );
+		const onUpdateOrder = jest.fn();
 
-		wrapper.find( 'input' ).simulate( 'change', {
-			target: {
-				value: '',
-			},
-		} );
+		render( <PageAttributesOrder onUpdateOrder={ onUpdateOrder } /> );
 
-		wrapper.find( 'input' ).simulate( 'change', {
-			target: {
-				value: ' ',
-			},
-		} );
+		const input = screen.getByRole( 'spinbutton' );
```

Comment:
TIL about [`spinbutton`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/spinbutton_role) role.

Is it possible to use `screen.getByLabelText` here? I think that would make tests more easily understandable.

Reply:
It is possible, of course. However, I believe the query by role is more precise (see [here](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-byrole-most-of-the-time)) and I'd actually expect to see a number field managing the page order. 

cc @ciampo @mirka thoughts?

Reply:
Sure, we should try and use a role selector whenever possible. I just wanted to note that it was hard for me to connect the `spinbutton` and Order input field. I had to check a11y tree in Chrome to make this connection 😅

Reply:
I guess I've been digging too much into a11y with the RTL migration nowadays 😅 But also, there's always something to learn about the WAI-ARIA specs 😉

Reply:
Best of both worlds!

```suggestion
		const input = screen.getByRole( 'spinbutton', { name: 'The label of the thing' } );
```

Reply:
Agreed the more specific one is better 👍 Addressed in 35179ad6c39c0d3aeeaef01abff642c9455f3525